src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -881,9 +881,7 @@ class VariantSampleMatrix[RPK, RK, T >: Null](val hc: HailContext, val metadata:
   def deleteVA(path: List[String]): (Type, Deleter) = vaSignature.delete(path)
 
   def dropSamples(): VariantSampleMatrix[RPK, RK, T] =
-    copy(sampleIds = IndexedSeq.empty[Annotation],
-      sampleAnnotations = IndexedSeq.empty[Annotation],
-      rdd = rdd.mapValues { case (va, gs) => (va, Iterable.empty[T]) })
+    copyAST(ast = FilterSamples(ast, Const(null, true, TBoolean)))
```

Comment:
You want `false` here, right?

Reply:
I know every FP lang does this but I dislike the term `filter`. I always have to lookup if its a `select` or a `reject`.